<launch>
    <arg name="use_camera" default="true"/>
    <arg name="use_arm" default="true"/>
    <arg name="use_base" default="false"/>
    <arg name="use_sim" default="true"/>
    <arg name="use_rviz" default="true"/>
    <arg name="base" default="kobuki"/> <!-- Options: create and kobuki -->

    <param name="use_camera" value="$(arg use_camera)"/>
    <param name="use_arm" value="$(arg use_arm)"/>
    <param name="use_sim" value="$(arg use_sim)"/>
    <param name="use_base" value="$(arg use_base)"/>
    

    <!-- Hardware -->
    <group unless="$(arg use_sim)">

        <group if="$(arg use_camera)">
            <include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
                <arg name="enable_pointcloud" value="false"/>
            </include>
        </group>

        <group if="$(arg use_base)" >
            <include file="$(find base_navigation)/launch/main_base.launch">
                <arg name="base" value="$(arg base)"/>
            </include>
        </group>

        <group if="$(eval arg('use_arm') or arg('use_camera'))">
            <include file="$(find locobot_control)/launch/dynamixel_controllers.launch"/>
        </group>

        <node name="calibration_tf_broadcaster" pkg="locobot_calibration"
              type="calibration_publish_transforms.py"/>

    </group>

    <!-- Gazebo -->
    <group if="$(arg use_sim)">
        
        <include file="$(find robot_arm)/launch/gazebo_locobot.launch">
            <arg name="base" value="$(arg base)"/>
        </include>

        <include file="$(find robot_arm)/launch/gazebo_locobot_control.launch"/>
        
        <node name="locobot_gazebo_no_pan_tilt" pkg="robot_arm" type="locobot_gazebo_no_pan_tilt" 
            respawn="true" output="screen"/> 

    </group>


    <!-- Common -->
    <node name="pyrobot_kinematics" pkg="pyrobot_bridge"
                  type="kinematics.py"/>

    
    <group if="$(eval arg('use_base') or arg('use_sim'))" >
        <node pkg="tf" type="static_transform_publisher" name="odom_map_broadcaster" args="0 0 0 0 0 0 /map /odom 100" />
        <!--include file="$(find base_navigation)/launch/move_base.launch">
            <arg name="use_map" value="false"/>
            <arg name="base" value="$(arg base)"/>
        </include-->
    </group>

    <include file="$(find robot_arm)/launch/moveit.launch">
        <arg name="use_rviz" value="$(arg use_rviz)"/>
    </include>
    
    <include file="$(find robot_arm)/launch/perception.launch"/>
    

</launch>
